packages/p2p/src/stores/my-profile-store.js

```diff
@@ -312,14 +313,16 @@ export default class MyProfileStore extends BaseStore {
 
     @action.bound
     handleToggle() {
+        this.root_store.general_store.setShouldShowRealName(!this.root_store?.general_store?.should_show_real_name);
```

Comment:
Why is this outside the request?

Reply:
Binding toggle switch to an API response is not ideal imho. The toggle should switch once when user clicks on it, and stay switched for valid API response and switch back to previous state for error response.

Reply:
IMHO it shouldn't switch back when there's an error, that's bad UI and UX. A slight delay in the switch is much better than allowing the user to change it and changing it back when there's an error.